src/tests/unit/auto/auto_ctput_test.cpp

```diff
@@ -0,0 +1,246 @@
+                            ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .WillByDefault(Return(gpuMockExeNetwork));
+
+        std::shared_ptr<ngraph::Function> simpleNetwork = ngraph::builder::subgraph::makeSingleConv();
+        ASSERT_NO_THROW(simpleCnnNetwork = InferenceEngine::CNNNetwork(simpleNetwork));
+    }
+};
+
+TEST_P(LoadNetworkWithCTPUTMockTest, CTPUTSingleDevLogicTest) {
+    std::vector<std::string> targetDevices;
+    Config config;
+    std::tie(targetDevices) = this->GetParam();
+
+    plugin->SetName("AUTO");
+    config.insert({{CONFIG_KEY(PERFORMANCE_HINT), InferenceEngine::PluginConfigParams::CUMULATIVE_THROUGHPUT}});
+
+    if (targetDevices.size() == 1) {
+        std::string targetDevice = targetDevices[0];
+        config.insert({InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, targetDevices[0]});
+        // Call single device logic and performance hint is THROUGHPUT
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(targetDevice),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(
+                                    ComparePerfHint(InferenceEngine::PluginConfigParams::THROUGHPUT))))
+            .Times(1);
+        // if target device only has GPU, no CPU helper to be called
+        if (targetDevice.find("GPU") != std::string::npos) {
+            EXPECT_CALL(*core,
+                        LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                    ::testing::Matcher<const std::string&>(CommonTestUtils::DEVICE_CPU),
+                                    ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+                .Times(0);
+        }
+    } else {
+        std::string targetDev;
+        for (auto& deviceName : targetDevices) {
+            targetDev += deviceName;
+            targetDev += ((deviceName == targetDevices.back()) ? "" : ",");
+        }
+        config.insert({InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, targetDev});
+        //Call MULTI logic
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>("MULTI:" + targetDev),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(1);
+        // no CPU helper to be called
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(CommonTestUtils::DEVICE_CPU),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(0);
```

Comment:
if its CPU help, it should check the third parameter to match perf hint as latency as well

Reply:
Updated